CommonLibSF/src/RE/A/Array.cpp

```diff
@@ -0,0 +1,196 @@
+#include "RE/A/Array.h"
+
+namespace RE
+{
+	namespace BSScript
+	{
+		Array::~Array()
+		{
+			dtor();
+		}
+
+		Array::Array(const TypeInfo* type_info, std::uint32_t initial_size)
+		{
+			ctor(type_info, initial_size);
+		}
+
+		Array* Array::ctor(const TypeInfo* type_info, std::uint32_t initial_size)
+		{
+			using func_t = decltype(&Array::ctor);
+			REL::Relocation<func_t> func{ ID::BSScript::Array::ctor };
+			return func(this, type_info, initial_size);
+		}
+
+		void Array::dtor()
+		{
+			using func_t = decltype(&Array::dtor);
+			REL::Relocation<func_t> func{ ID::BSScript::Array::dtor };
+			return func(this);
+		}
+
+		[[nodiscard]] auto Array::operator[](size_type a_pos)
+			-> reference
+		{
+			assert(a_pos < size());
+			return *(elements.data() + static_cast<std::ptrdiff_t>(a_pos));  // TODO: add operator[] to BSTArray
```

Comment:
Fix this to be sizeof(variable) *pos

Reply:
Never mind, elements.data() is a pointer to an array of pointers, not an array of the data type